entity-framework/core/modeling/relationships/conventions.md

```diff
@@ -0,0 +1,450 @@
+    Skip navigations:
+      Posts (ICollection<Post>) CollectionPost Inverse: Tags
+    Keys:
+      Id PK
+  EntityType: PostTag (Dictionary<string, object>) CLR Type: Dictionary<string, object>
+    Properties:
+      PostsId (no field, int) Indexer Required PK FK AfterSave:Throw
+      TagsId (no field, int) Indexer Required PK FK Index AfterSave:Throw
+    Keys:
+      PostsId, TagsId PK
+    Foreign keys:
+      PostTag (Dictionary<string, object>) {'PostsId'} -> Post {'Id'} Cascade
+      PostTag (Dictionary<string, object>) {'TagsId'} -> Tag {'Id'} Cascade
+    Indexes:
+      TagsId
+```
+
+And translates to the following database schema when using SQLite:
+
+```sql
+CREATE TABLE "Posts" (
+    "Id" INTEGER NOT NULL CONSTRAINT "PK_Posts" PRIMARY KEY AUTOINCREMENT);
+
+CREATE TABLE "Tag" (
+    "Id" INTEGER NOT NULL CONSTRAINT "PK_Tag" PRIMARY KEY AUTOINCREMENT);
+
+CREATE TABLE "PostTag" (
+    "PostsId" INTEGER NOT NULL,
+    "TagsId" INTEGER NOT NULL,
+    CONSTRAINT "PK_PostTag" PRIMARY KEY ("PostsId", "TagsId"),
+    CONSTRAINT "FK_PostTag_Posts_PostsId" FOREIGN KEY ("PostsId") REFERENCES "Posts" ("Id") ON DELETE CASCADE,
+    CONSTRAINT "FK_PostTag_Tag_TagsId" FOREIGN KEY ("TagsId") REFERENCES "Tag" ("Id") ON DELETE CASCADE);
+
+CREATE INDEX "IX_PostTag_TagsId" ON "PostTag" ("TagsId");
+```
+
+## Indexes
+
+By convention, EF creates a [database index](xref:core/modeling/indexes) for the property or properties of a foreign key. The type of index created is determined by:
+
+- The cardinality of the relationship
+- Whether the relationship is optional or required
+- The number of properties that make up the foreign key
+
+For a [one-to-many relationship](xref:core/modeling/relationships/one-to-many), a straightforward index is created by convention. The same index is created for optional and required relationships. For example, on SQLite:
+
+```sql
+CREATE INDEX "IX_Post_BlogId" ON "Post" ("BlogId");
+```
+
+Or on SQL Server:
+
+```sql
+CREATE INDEX [IX_Post_BlogId] ON [Post] ([BlogId]);
+```
+
+For an required [one-to-one relationship](xref:core/modeling/relationships/one-to-one), a unique index is created. For example, on SQLite:
+
+```sql
+CREATE UNIQUE INDEX "IX_Author_BlogId" ON "Author" ("BlogId");
+```
+
+Or on SQL Sever:
+
+```sql
+CREATE UNIQUE INDEX [IX_Author_BlogId] ON [Author] ([BlogId]);
+```
+
+For optional one-to-one relationships, the index created on SQLite is the same:
+
+```sql
+CREATE UNIQUE INDEX "IX_Author_BlogId" ON "Author" ("BlogId");
+```
+
+However, on SQL Server, an `IS NOT NULL` filter is added to better handle null foreign key values. For example:
+
+```sql
+CREATE UNIQUE INDEX [IX_Author_BlogId] ON [Author] ([BlogId]) WHERE [BlogId] IS NOT NULL;
+```
+
+For composite foreign keys, an index is created covering all the foreign key columns. For example:
+
+```sql
+CREATE INDEX "IX_Post_ContainingBlogId1_ContainingBlogId2" ON "Post" ("ContainingBlogId1", "ContainingBlogId2");
+```
+
```

Comment:
We should also mention that EF doesn't create indexes for properties already covered by other indexes or keys